pyseas/maps/bivariate.py

```diff
@@ -0,0 +1,239 @@
+
+import numpy as np
+from matplotlib.colors import Normalize, LogNorm
+from matplotlib import cm
+import matplotlib.pyplot as plt
+import warnings
+from . import core
+from .. import props
+
+
+class BivariateColormap:
+
+    log_x = None
+    log_y = None
+
+    def __call__(self, X, Y, alpha=None):
+        """Return color for the pair of values X and Y.
+
+        Parameters
+        ----------
+        X, Y : float or np.array of float
+            Each pair of X, Y values is mapped to a color value.
+            The data should be in the range [0.0, 1.0].
+        alpha : float or np.array or float
+            A scalar in the range [0.0, 1.0] that specifies the 
+            opacity at each X, Y pair. If alpha is an array, it 
+            must match the shape of X and Y.   
+
+        Return
+        ------
+        np.array
+            The shape will be np.shape(X) + (4,) to accommodate RGBA
+            information.       
+        """
+
+
+class TransparencyBivariateColormap(BivariateColormap):
+    """Bivariate colormap combining standard colormap with transparency
+
+    The first value (`X`) is looked up in `base_cmap` and the result is
+    multiplied by the second value ('Y').
+
+    Parameters
+    ----------
+    base_cmap : matplotlib Colormap
+    log_x, log_y : bool, optional
+    """
+
+    def __init__(self, base_cmap, log_x=None, log_y=None):
+        self.base_cmap = base_cmap
+        self.log_x = log_x
+        self.log_y = log_y
+
+    # TODO: should make this parameterizable
+    def __call__(self, X, Y, alpha=None):
+        colors = self.base_cmap(X)
+        colors[..., 3] *= Y
+        if alpha is not None:
```

Comment:
Do we want an example in Examples.ipynb using the alpha parameter? It's not listed under the parameters in the docstring. Is it just a single number from 0 to 1?

Reply:
I don't have a good example for it handy, but it's there to match an existing interface as I recall. I'll document it and call that good for now.

Reply:
Looking at this, alpha is documented in the supercalss (along with the rest of the __call__ arguments.  In Python 3, that docstring will get inherited by the subclasses. Since they all have the same interface, I think this can be left alone.